### PR TITLE
[ONNX] Fix scalar elements in op.Concat

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98509
* #97494

op.Concat wrongly concatenated scalar int, and it would raise errors in ORT. However, we didn't see this bug until SegFault was fixed.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py [hunk 1]
```diff
@@ -514,11 +514,7 @@ def forward(self, x):
         )
 
     @pytorch_test_common.skip_dynamic_fx_test(
-        "1. flaky test: https://github.com/microsoft/onnx-script/issues/523. "
-        "Fixed in ORT==1.15"
-        "2. [ONNXRuntimeError] : 1 : FAIL : Non-zero status code returned while "
-        "running Concat node. Name:'Concat_203' Status Message: concat.cc:104 "
-        "PrepareForCompute Cannot concatenate scalars"
+        "flaky test: https://github.com/microsoft/onnx-script/issues/523. Fixed in ORT==1.15"
```
Comment (BowenBao) [inline test/onnx/test_fx_to_onnx_with_onnxruntime.py:517]: Do you know if there is a ort-nightly version we could pin in CI?

Consider adding a skip based on ORT version too.

**Review by BowenBao — APPROVED** [review]

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py (hunk 1 above)
Comment (titaiwangms) [reply test/onnx/test_fx_to_onnx_with_onnxruntime.py:517]: It seems we can get it from AzureDevOps, and as long as AzureDevOps doesn't crash, we should be good. But I feel risky...

Comment (titaiwangms) [reply test/onnx/test_fx_to_onnx_with_onnxruntime.py]: Added skip_unsupported_ort_min_version. @justinchuby LMK if I use packaging.version.parse correctly!

File: test/onnx/pytorch_test_common.py [hunk 2]
```diff
@@ -157,6 +158,24 @@ def wrapper(self, *args, **kwargs):
     return skip_dec
 
 
+def skip_unsupported_min_ort_version(min_opset_version: str):
```
Comment (justinchuby) [inline test/onnx/pytorch_test_common.py:161]:
```suggestion
def skip_unsupported_min_ort_version(reason: str, min_opset_version: str):
```

We should still specify reason

Comment (justinchuby) [reply test/onnx/pytorch_test_common.py:161]: opset version was a typo?

File: test/onnx/pytorch_test_common.py [hunk 3]
```diff
@@ -157,6 +158,24 @@ def wrapper(self, *args, **kwargs):
...
+    def skip_dec(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if (
+                packaging.version.parse(self.ort_version).release
+                < packaging.version.parse(min_opset_version).release
+            ):
+                raise unittest.SkipTest(
+                    f"Unsupported onnxruntime version: {self.ort_version} < {min_opset_version}"
```
Comment (justinchuby) [inline test/onnx/pytorch_test_common.py:170]: I would make it clear on what is not supported. This sentence can be read both ways (too new or too old) Maybe something like "ONNX Runtime version ... is older than required version ..." ?

File: test/onnx/test_fx_to_onnx_with_onnxruntime.py (hunk 1 above)
Comment (justinchuby) [reply test/onnx/test_fx_to_onnx_with_onnxruntime.py]: packaging.version lgtm!

File: test/onnx/pytorch_test_common.py [hunk 4]
```diff
@@ -158,16 +158,17 @@ def wrapper(self, *args, **kwargs):
...
         def wrapper(self, *args, **kwargs):
             if (
                 packaging.version.parse(self.ort_version).release
-                < packaging.version.parse(min_opset_version).release
+                < packaging.version.parse(version).release
             ):
                 raise unittest.SkipTest(
-                    f"Unsupported onnxruntime version: {self.ort_version} < {min_opset_version}"
+                    f"ONNX Runtime version: {version} is older than required version {version}."
+                    f"This reason we skip it: {reason}."
```
Comment (justinchuby) [inline test/onnx/pytorch_test_common.py:171]:
```suggestion
                    f"Reason: {reason}."
```

for simplicity

File: test/onnx/pytorch_test_common.py (hunk 2 above)
Comment (titaiwangms) [reply test/onnx/pytorch_test_common.py]: Done

**Pull request closed without merging**